core/azure-core/src/main/java/com/azure/core/util/polling/OperationStatus.java

```diff
@@ -0,0 +1,70 @@
+package com.azure.core.util.polling;
+
+import java.util.Objects;
+
+public class OperationStatus {
```

Comment:
Moving it into separate class since now, it will have its own POJO like structure as we are supporting OTHER state and must have String value for custom other state.